lib/delayed/active_job_adapter.rb

```diff
@@ -17,7 +17,7 @@ def enqueue_at(job, timestamp)
     private
 
     def _enqueue(job, opts = {})
-      if job.class.respond_to?(:enqueue_after_transaction_commit) && job.class.enqueue_after_transaction_commit
+      if job.class.respond_to?(:enqueue_after_transaction_commit) && job.class.enqueue_after_transaction_commit == :always
```

Comment:
This isn't the perfect solution to this problem. The implementation in `ActiveJob::EnqueueAfterTransactionCommit`  does allow for unset (or none `:always`/`:never`) values which then gets delegated to `job_adapter.enqueue_after_transaction_commit?`. Given the job adapter will always be delayed, if a none `:always` value is passed in, it will always be `false`

Reply:
Yup, that was my read as well.

But, gosh, that Rails API seems like it's going to cause such confusion. I.e. you could set `def self.enqueue_after_transaction_commit; true; end` and then it goes down the `false` codepath because nothing about the API itself checks for completeness on `:always`/`:never`/`:default`. 🤔

Reply:
Yep, feels like there is a missing validation on the value you are passing in.

Reply:
Heh, @warmerzega https://github.com/rails/rails/pull/53375